Modules/Core/include/selxNetworkBuilder.hxx

```diff
@@ -265,16 +267,23 @@ NetworkBuilder< ComponentList >::PropagateConnectionsWithUniqueComponents()
 
           auto               acceptingComponent = this->m_ComponentSelectorContainer[ acceptingComponentName ]->GetComponent();
           const unsigned int beforeCriteria    = this->m_ComponentSelectorContainer[ componentName ]->NumberOfComponents();
-          this->m_Logger.Log( LogLevel::DBG, "Propagating 'ProvidingInterface' properties from '{0}' to {2} components at '{1}' ... ", componentName, acceptingComponentName, beforeCriteria );
+          this->m_Logger.Log( LogLevel::DBG, "Trying to narrow candidates for '{0}' based on the connections that '{0}' provides and '{1}' accepts ... ", componentName, acceptingComponentName );
           this->m_ComponentSelectorContainer[ componentName ]->RequireProvidingInterfaceTo( acceptingComponent, interfaceCriteria );
           const unsigned int afterCriteria = this->m_ComponentSelectorContainer[ componentName ]->NumberOfComponents();
-          this->m_Logger.Log( LogLevel::DBG, "Propagating 'ProvidingInterface' properties from '{0}' to {2} components at '{1}' ... Done. Reduced '{1}' to {3} components", componentName, acceptingComponentName, beforeCriteria, afterCriteria );
+
+          if( afterCriteria == 0 ) {
+            this->m_Logger.Log( LogLevel::DBG, "Trying to narrow candidates for '{0}' based on the connections that '{0}' provides and '{1}' accepts ... Error: No connections can be made from {0} to {1}.", componentName, acceptingComponentName, beforeCriteria, afterCriteria );
+            throw std::runtime_error("No connections can be made from " + componentName + " to " + acceptingComponentName + ".");
+          }
 
           if( beforeCriteria > afterCriteria )
           {
-            anySelectionNarrowed = true;
```

Comment:
You should not remove the "anySelectionNarrowed = true", since this stops the while loop after one run always, while it might need to run multiple times to fully narrow down all selections.